src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2196,7 +2199,42 @@ void SessionImpl::populateAdditionalTrackers()
     {
         tracker = tracker.trimmed();
         if (!tracker.isEmpty())
-            m_additionalTrackerList.append({tracker.toString()});
+        {
+            // Scan the list of trackers to check if there are links to .txt files
+            // and add their content to the list of trackers
+            if (tracker.startsWith(u"http", Qt::CaseInsensitive) && tracker.endsWith(u".txt", Qt::CaseInsensitive))
+            {
+                // Download the content of the .txt file
+                QNetworkAccessManager manager;
+                QNetworkRequest const request(QUrl(tracker.toString()));
+                QNetworkReply *reply = manager.get(request);
+
+                // Wait for the reply
+                QObject::connect(reply, &QNetworkReply::finished, this, [reply, this]()
+                {
+                    const QString trackerContent = QString::fromUtf8(reply->readAll());
+                    if (!trackerContent.isEmpty())
+                    {
+                        // Add the trackers to the list
+                        auto trackerList = trackerContent.split(QChar(u'\n'), Qt::SkipEmptyParts);
+                        for (const auto &tracker: asConst(trackerList))
+                        {
+                            m_additionalTrackerList.append({tracker.trimmed()});
+                        }
+                    }
+                    reply->deleteLater();
+                });
+
+                // Wait for the reply to finish
```

Comment:
Doesn't seem like a good idea.

Reply:
May I ask you why do you think it's bad? I've used this technique already in other cases and it seemed to work good without memory leaks, but I'm actually not ax expert in Qt development, if there's a better way I'm curios to learn.

Reply:
Creating a local event loop inside another one makes the flow of the program even more confusing, while making a false impression of its straightforwardness, which can lead to difficult-to-diagnose errors. After encountering such errors a couple of times, we preferred to get rid of such local event loops (although we never created them directly, but there are some Qt functions doing this behind the scene, e.g. `Dialog::exec`).

Reply:
That makes sense. Okay I'll try to refactor this code.